### PR TITLE
initial configuration and unit test for CTA button added

### PR DESCRIPTION
**Pull request by gatinhap**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +26,9 @@
     "styled-components": "^6.0.8"
   },
   "devDependencies": {
+    "@testing-library/jest-dom": "^6.4.2",
+    "@testing-library/react": "^14.2.2",
+    "@types/jest": "^29.5.12",
```
Comment (michal-karbasz) [inline package.json:31]: why do you need this type package?

File: src/components/callToActionButton/CallToActionButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const { getByText } = render(
+    <ThemeProvider theme={theme}>
+      <CallToActionButton handleClick={onClickMock}>
+        click me
+      </CallToActionButton>
+    </ThemeProvider>,
+  );
+
+  fireEvent.click(getByText("click me"));
+  expect(onClickMock).toHaveBeenCalledTimes(1);
```
Comment (michal-karbasz) [inline src/components/callToActionButton/CallToActionButton.test.tsx:18]: that's up to the team convention, but FYI we tend to put a new line for each expression that doesn't fit into a block. In a React component, it looks like this, e.g.:

react hooks block
{new line}
react query hooks
{new line}
custom hooks
{new line}
event handlers
{new line}

and so on - we're putting similar things visually together and split the blocks with a new line.

In tests it works the same

render
{new line}
constants
{new line}
event
{new line}
expect


and so on. Also, you can use a new line whenever your eyes have a hard time reading expressions that belong to the same group.

File: src/components/callToActionButton/CallToActionButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { expect, vi } from "vitest";
+import CallToActionButton from "./CallToActionButton.tsx";
+import { fireEvent, render } from "@testing-library/react";
+import { theme } from "../../theme/theme.ts";
+import { ThemeProvider } from "styled-components";
+
+test("CTA button onClick event calls function when button is clicked", () => {
+  const onClickMock = vi.fn();
```
Comment (michal-karbasz) [inline src/components/callToActionButton/CallToActionButton.test.tsx:8]: in general mocks are global for entire test file and can be put right after the imports

File: src/components/callToActionButton/CallToActionButton.test.tsx
Comment (michal-karbasz) [inline src/components/callToActionButton/CallToActionButton.test.tsx:1]: I like to keep at least unit tests colocated with the file that's tested. No brainer + I'm not a fan of duplicating folder structures for tests. 
With integration tests we tend to create a separate folder called "integration" next to the page that's tested.

File: package.json (hunk 1 above)
Comment (gatinhap) [reply package.json:31]: While trying to write and run this test, one of the tutorials suggested installing this package to solve a TS error in `vite.config.js`.
Screenshot 2024-04-11 at 14 41 01

File: src/components/callToActionButton/CallToActionButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const { getByText } = render(
+    <ThemeProvider theme={theme}>
+      <CallToActionButton handleClick={onClickMock}>
+        click me
+      </CallToActionButton>
+    </ThemeProvider>,
```
[the remark this answers is not in the document]
Comment (gatinhap) [reply src/components/callToActionButton/CallToActionButton.test.tsx:14]: What should I do here? There is a comma🤔

File: package.json [hunk 5]
```diff
@@ -7,11 +7,14 @@
...
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "test": "vitest",
+    "coverage": "vitest run --coverage"
   },
   "dependencies": {
     "@hookform/error-message": "^2.0.1",
     "@tanstack/react-query": "^5.14.2",
+    "jsdom": "^24.0.0",
```
[the remark this answers is not in the document]
Comment (michal-karbasz) [reply package.json:17]: sure, but there are dependencies and dev-dependencies in package.json - whatever package is not necessary during the build of the project should go to dev-dependencies. Usually everything testing-related is a dev dep

File: package.json (hunk 1 above)
Comment (michal-karbasz) [reply package.json:31]: hm, defineConfig accepts either an object or a function, maybe you can try with the function approach. We don't use that package and it works. Minor issue but let's try

File: src/components/callToActionButton/CallToActionButton.test.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (michal-karbasz) [reply src/components/callToActionButton/CallToActionButton.test.tsx:14]: can you delete it? Why do you need the comma after tag? :D

File: src/components/formElements/InputField.test.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+describe("Input field attributes are accepted", () => {
+  it("renders placeholder text", () => {
+    render(
+      <ThemeProvider theme={theme}>
+        <InputField placeholder={"placeholder text"} />
+      </ThemeProvider>,
+    );
+
+    const inputElement: HTMLInputElement =
```
Comment (michal-karbasz) [inline src/components/formElements/InputField.test.tsx:15]: hm, that type should not be necessary

File: package.json (hunk 1 above)
Comment (gatinhap) [reply package.json:31]: With the function approach it worked and the TS error is not showing anymore so I removed @types/jest library

File: src/components/callToActionButton/CallToActionButton.test.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (gatinhap) [reply src/components/callToActionButton/CallToActionButton.test.tsx:15]: I'm not adding it but it was prettier adding trailing comma after each save. I had to make `.prettierrc.json` file and add there `{
  "trailingComma": "none"
}`. Now, it's not showing up anymore.

File: src/components/formElements/InputField.test.tsx (hunk 6 above)
Comment (gatinhap) [reply src/components/formElements/InputField.test.tsx:15]: I added it to get rid of TS error below:

Screenshot 2024-04-29 at 16 07 42.

However, while it's added like this: `const inputElement: HTMLInputElement = screen.getByTestId("input");`

I get another TS error😅:

Screenshot 2024-04-29 at 16 10 13

Do you have any suggestion how I could solve it?

File: src/components/scrollToTop/ScrollToTop.tsx [hunk 7]
```diff
@@ -5,7 +5,7 @@ const ScrollToTop = () => {
     window.scrollTo({ top: 0, left: 0, behavior: "smooth" });
   };
 
-  return <ScrollToTopStyled onClick={scrollUp} />;
+  return <ScrollToTopStyled role={"button"} onClick={scrollUp} />;
```
Comment (michal-karbasz) [inline src/components/scrollToTop/ScrollToTop.tsx:8]: superminor but FYI it's our code convention to write string props values like this `role="button"`

File: .prettierrc.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+  "trailingComma": "none"
+}
```
Comment (michal-karbasz) [inline .prettierrc.json:3]: :)

File: src/components/formElements/InputField.test.tsx (hunk 6 above)
Comment (michal-karbasz) [reply src/components/formElements/InputField.test.tsx:15]: inputElement.files?[0] should do this. TS is right - there's no certainty that files were uploaded.

**Review by michal-karbasz — APPROVED** [review]

**Pull request merged**